react/src/components/QrReaderOverlay/OrReaderOverlay.test.tsx

```diff
@@ -0,0 +1,332 @@
+
+  // Open QROverlay
+  const qrButton = await screen.findByRole("button", { name: /scan qr code/i });
+  await user.click(qrButton);
+
+  // Find Box
+  const findBoxButton = await screen.findByRole("button", { name: /find/i });
+  await user.type(screen.getByRole("textbox"), "123456");
+  await user.click(findBoxButton);
+
+  // error message appears
+  expect(await screen.findByText(/Box not found for this label/i)).toBeInTheDocument();
+  // QrOverlay stays open
+  expect(screen.getByRole("button", { name: /find/i })).toBeInTheDocument();
+});
+
+const queryFindBox = {
+  request: {
+    query: BOX_DETAILS_BY_LABEL_IDENTIFIER_QUERY,
+    variables: {
+      labelIdentifier: "123456",
+    },
+  },
+  result: {
+    data: {
+      box: generateMockBox({ labelIdentifier: "123456" }),
+    },
+  },
+};
+
+it("3.4.1.3 - Mobile: Enter valid box identifier and click on Find button", async () => {
+  const user = userEvent.setup();
+  mockImplementationOfQrReader(mockedQrReader, "BoxAssociatedWithQrCode");
+  // mock scanning a QR code
+  render(<HeaderMenuContainer />, {
+    routePath: "/bases/:baseId",
+    initialUrl: "/bases/1",
+    mocks: [queryFindBox],
+    additionalRoute: "/bases/1/boxes/123456",
+  });
+
+  // Open QROverlay
+  const qrButton = await screen.findByRole("button", { name: /scan qr code/i });
+  await user.click(qrButton);
+
+  // Find Box
+  const findBoxButton = await screen.findByRole("button", { name: /find/i });
+  await user.type(screen.getByRole("textbox"), "123456");
+  await user.click(findBoxButton);
+
+  // Click a button to trigger the event of scanning a QR-Code in mockImplementationOfQrReader
+  expect(await screen.findByRole("heading", { name: "/bases/1/boxes/123456" })).toBeInTheDocument();
+});
+
+const queryFindBoxFromOtherOrg = {
+  request: {
+    query: BOX_DETAILS_BY_LABEL_IDENTIFIER_QUERY,
+    variables: {
+      labelIdentifier: "123456",
+    },
+  },
+  result: {
+    data: {
+      box: null,
+    },
+    errors: [new GraphQLError("Error!", { extensions: { code: "FORBIDDEN" } })],
+  },
+};
+
+it("3.4.1.4 - Mobile: Enter valid box identifier but not from the user bases (unauthorized) and click on Find button", async () => {
+  const user = userEvent.setup();
+  mockImplementationOfQrReader(mockedQrReader, "NoBoxAssociatedWithQrCode");
+  // mock scanning a QR code
+  render(<HeaderMenuContainer />, {
+    routePath: "/bases/:baseId",
+    initialUrl: "/bases/1",
+    mocks: [queryFindBoxFromOtherOrg],
+    additionalRoute: "/bases/1/boxes/123456",
+  });
+
+  // Open QROverlay
+  const qrButton = await screen.findByRole("button", { name: /scan qr code/i });
+  await user.click(qrButton);
+
+  // Find Box
+  const findBoxButton = await screen.findByRole("button", { name: /find/i });
+  await user.type(screen.getByRole("textbox"), "123456");
+  await user.click(findBoxButton);
+
+  // error message appears
+  expect(
+    await screen.findByText(/You don't have permission to access this box/i),
```

Comment:
https://docs.google.com/spreadsheets/d/1sDhSsaVwNxAhGn1VYACDPVepZpvlwTkgy9nmeTaRy2Q/edit?pli=1&disco=AAAApSHIVhs

message should be: "You don't have permission to access Box $identifier"

Reply:
I talked to Roanna about this and we change Box $identifier to this box. I updated it in the Test Plan

Reply:
done